u-service-hello/src/main/java/com/gft/first/microservice_consul_1/MicroserviceConsul1Application.java

```diff
@@ -42,42 +42,66 @@ public String home() {
         return new ResponseEntity<>(message, HttpStatus.OK);
     }
 
-    @RequestMapping(value = "/api/inspect")
-    public String apiInspect(@RequestParam(value = "delay", required = false) Integer delay) {
+    @RequestMapping(value = "/api/inspect", method= RequestMethod.GET )
+    public String apiInspectCsv(HttpServletResponse response,
+                                                  @RequestParam(value = "delay", required = false) Integer delay,
+                                                  @RequestParam(value = "format", required = false) String format) throws IOException {
+
+        if (format != null && format.equals("Fcsv")) {
+            response.addHeader("Content-disposition", "attachment;filename=myfilename.csv");
+            response.setContentType("txt/plain");
+            response.getOutputStream().write(getHostInfo("csv", delay).getBytes());
+            response.getOutputStream().flush();
+        }
+
+        return getHostInfo(format, delay);
+    }
 
+    private String getHostInfo(String format, Integer delay) {
         InetAddress ip = null;
         String hostname = null;
         StringBuffer info =  new StringBuffer();
+    try{
 
-        try {
-            if(delay != null){
-                LOGGER.info("Sleep for : " +  delay + "ms");
-                info.append("Your app was call with delay<b> " + delay + "ms</b></br>");
+        callThreadSleep(delay, info);
```

Comment:
cattThreadSleep shouldn't be part of getHostInfo
they are independent of each other - single responsibility
they are sequential but independant
please move it one level up